src/main/java/org/apache/hadoop/util/LineReader.java

```diff
@@ -240,12 +246,19 @@ private int readDefaultLine(Text str, int maxLineLength, int maxBytesToConsume)
                 appendLength = maxLineLength - txtLength;
             }
             if (appendLength > 0) {
+                int newTxtLength = txtLength + appendLength;
+                if (str.getBytes().length < newTxtLength && Math.max(newTxtLength, txtLength << 1) > MAX_ARRAY_SIZE) {
```

Comment:
Do we need to worry about overflow of `txtLength << 1`? Same question for the one below

Reply:
This is to simulate the behavior of Text.setCapacity() method. It would try to resize Text.bytes to Math.max(newTxtLength, txtLength << 1) when keepData is true, which is the case for append().

Reply:
In such case Math.max(newTxtLength, txtLength << 1) will be newTxtLength which is <= maxLineLength and won't overflow. This is the same logic in Text.setCapacity().